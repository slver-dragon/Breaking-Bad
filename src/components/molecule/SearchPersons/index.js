import React from "react";
import Button from "../../atom/Button";
import lens from "../../../assets/img/search.png";
import cross from "../../../assets/img/cross.png";
import style from "./style.module.scss";

const SearchPersons = ({ search }) => {
  let request = React.createRef();
  const discardSearch = () => {
    search("");
    request.current.value = "";
  };
  const pressEnter = (event) => {
    if (event.keyCode === 13) search(request.current.value);
  };
  return (
    <div className={style.search}>
      <label className={style.container}>
        <span onClick={() => discardSearch()}>
          <Button img={cross} />
        </span>
        <input
          type="text"
          placeholder="Поиск"
          ref={request}
          onKeyDown={(event) => {
            pressEnter(event);
          }}
        />
        <span onClick={() => search(request.current.value)}>
          <Button img={lens} text="Найти" />
        </span>
      </label>
    </div>
  );
};

export default SearchPersons;
