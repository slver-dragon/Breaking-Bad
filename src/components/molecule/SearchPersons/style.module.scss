@import "../../../assets/styles/style";
.search {
  display: flex;
  justify-content: space-between;
  align-items: center;
  @include mainContainer;
  @include shadowBox;
  @include searchField;
  padding: 1rem;
  input {
    width: 90%;
    &:focus {
      outline: none;
    }
  }
}
