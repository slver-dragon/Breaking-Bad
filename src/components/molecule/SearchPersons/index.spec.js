import React from "react";
import { mount, shallow } from "enzyme";
import SearchPersons from ".";

describe("Testing molecule SearchPersons:", () => {
  let component = null;
  const onBtnClick = jest.fn();
  const onClickEnter = jest.fn();
  const setComponent = (props) => mount(<SearchPersons {...props} />);

  test("is rendering", () => {
    component = setComponent();
    expect(component).toMatchSnapshot();
  });

  test("discard search value", () => {
    component = setComponent({ search: onBtnClick });
    component.find("input").simulate('change', { target: { value: 'Walter White' } });
    component.find("#Clear").simulate("click");
    expect(onBtnClick).toHaveBeenCalled();
  });

  test("search run", () => {
    component = setComponent({ search: onBtnClick });
    component.find("input").simulate('change', { target: { value: 'Walter White' } });
    component.find("#Search").simulate("click");
    expect(onBtnClick).toHaveBeenCalled();
    component = setComponent({ search: onClickEnter });
    component.find("input").simulate("keydown", { keyCode: 13});
    expect(onClickEnter).toHaveBeenCalled();
  });
});
