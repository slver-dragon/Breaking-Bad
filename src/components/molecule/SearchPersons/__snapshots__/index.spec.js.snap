// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing molecule SearchPersons: is rendering 1`] = `
<SearchPersons>
  <div
    className="search"
  >
    <label
      className="container"
    >
      <span
        id="Clear"
        onClick={[Function]}
      >
        <Button
          img="test-file-stub"
        >
          <button
            className="button"
          >
            <img
              alt=""
              src="test-file-stub"
            />
            <span />
          </button>
        </Button>
      </span>
      <input
        onKeyDown={[Function]}
        placeholder="Поиск"
        type="text"
      />
      <span
        id="Search"
        onClick={[Function]}
      >
        <Button
          img="test-file-stub"
          text="Найти"
        >
          <button
            className="button"
          >
            <img
              alt=""
              src="test-file-stub"
            />
            <span>
              Найти
            </span>
          </button>
        </Button>
      </span>
    </label>
  </div>
</SearchPersons>
`;
